File: jars/HerbPack/src/main/java/herb/herb/commands/TraceCommand.java
package herb.herb.commands;

import herb.herb.utilitys.GlobalArgus;
import org.bukkit.Bukkit;
import org.bukkit.Material;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.command.TabExecutor;
import org.bukkit.entity.Player;
import org.bukkit.inventory.Inventory;
import org.bukkit.inventory.ItemStack;
import org.bukkit.inventory.meta.SkullMeta;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

import static org.bukkit.Bukkit.getLogger;

public class TraceCommand implements TabExecutor {
    public Inventory UI(int i) {
        @NotNull Inventory inv = Bukkit.createInventory(null, 36, new GlobalArgus().TracerName);
        inv.clear();
        int i_ = (i - 1) * 36;
        Bukkit.getOnlinePlayers().forEach(p -> {
            if (i_ == 0) {
                for(int o=36;o != 0;o--){
                    ItemStack head = new ItemStack(Material.PLAYER_HEAD);
                    SkullMeta headM = (SkullMeta) head.getItemMeta();
                    headM.setOwningPlayer(p);
                    headM.setPlayerProfile(p.getPlayerProfile());
                    headM.setDisplayName(p.getClientBrandName());
                    head.setItemMeta(headM);
                    inv.setItem(inv.firstEmpty(), head);
                }
            }
        });
        return inv;
    }
    @Override
    public boolean onCommand(@NotNull CommandSender sender, @NotNull Command command, @NotNull String label, @NotNull String[] args) {
        if(sender instanceof Player) {
            Player p = (Player) sender;
            if (args[0].equals(null)) {
                p.openInventory(UI(1));
            }else {
                p.openInventory(UI(Integer.parseInt(args[0])));
            }
            return true;
        }else {
            getLogger().warning("플레이어 전용 명령어 입니다!");
            return false;
        }
    }

    @Override
    public @Nullable List<String> onTabComplete(@NotNull CommandSender sender, @NotNull Command command, @NotNull String alias, @NotNull String[] args) {
        if (command.getName().equals("tracer")) {
            if (args.length == 1) {
                List numList = new ArrayList();
                for (int i=1; i>=Bukkit.getOnlinePlayers().size() / 36; i++){
                    numList.add(i);
                }
                return numList;
            }
        }
        return null;
    }
}

File: jars/HerbPlugin/main/java/herb/herb/commands/TraceCommand.kt
package herb.herb.commands

import org.bukkit.command.TabExecutor
import herb.herb.Argus.GlobalArgus
import org.bukkit.entity.Player
import org.bukkit.Bukkit
import org.bukkit.Material
import org.bukkit.inventory.ItemStack
import org.bukkit.inventory.Inventory
import org.bukkit.inventory.meta.SkullMeta
import herb.herb.Events.TraceEvent
import net.kyori.adventure.text.Component
import org.bukkit.command.Command
import org.bukkit.command.CommandSender

class TraceCommand constructor() : TabExecutor {
    fun UI(): Inventory {
        val pls: Collection<Player> = Bukkit.getOnlinePlayers()
        val inv: Inventory = Bukkit.createInventory(null, pls.size, GlobalArgus.TracerName)
        inv.clear()
        pls.forEach({ p: Player ->
            val head: ItemStack = ItemStack(Material.PLAYER_HEAD)
            val headM: SkullMeta = head.getItemMeta() as SkullMeta
            headM.setOwningPlayer(p)
            headM.setPlayerProfile(p.getPlayerProfile())
            headM.displayName(Component.text((p.getClientBrandName())!!))
            head.setItemMeta(headM)
            inv.setItem(inv.firstEmpty(), head)
        })
        return inv
    }

    public override fun onCommand(sender: CommandSender, command: Command, label: String, args: Array<String>): Boolean {
        GlobalArgus.pm.registerEvents(TraceEvent(), (GlobalArgus.plugin)!!)
        if (sender is Player) {
            sender.openInventory(UI())
            GlobalArgus.pm.registerEvents(TraceEvent(), GlobalArgus.plugin)
            return true
        } else {
            Bukkit.getLogger().warning("플레이어 전용 명령어 입니다!")
            return false
        }
    }

    public override fun onTabComplete(sender: CommandSender, command: Command, alias: String, args: Array<String>): List<String>? {
        return null
    }
}